internal/gxexec/execute.go

```diff
@@ -0,0 +1,67 @@
+//go:build gx
+
+package gxexec
+
+import (
+	"strings"
+
+	"github.com/janpfeifer/gonb/common"
+	"github.com/janpfeifer/gonb/internal/goexec"
+	"github.com/janpfeifer/gonb/internal/kernel"
+
+	"github.com/janpfeifer/gonb/internal/jpyexec"
+	"k8s.io/klog/v2"
+)
+
+// Executes the %%gx cell magic command.
+//
+// Supports compiling and generating binding for gx functions
+// so they can be used from Go code in the notebook.
+func ExecuteCell(msg kernel.Message, goExec *goexec.State, lines []string) error {
+	if klog.V(2).Enabled() {
+		klog.Infof("GX cell: %q", strings.Join(lines, "\n"))
+	}
+
+	stdout := kernel.NewJupyterStreamWriter(msg, kernel.StreamStdout)
+
+	// Find the last line that is gx code, ie before we see %go or %%
+	last_gx_line := -1
+	for i, line := range lines {
+		if line == "%go" || line == "%%" {
+			break
+		}
+		last_gx_line = i
+	}
+
+	// Echo gx code
+	// We'll actually want to compile Gx code here, and create some kind of
+	// declaration 'memory' that will be used to merge multiple GX cells together.
+	// For now we just print it.
+	err := jpyexec.New(msg, "printf", append([]string{"%s\n"}, lines[1:last_gx_line]...)...).
```

Comment:
Here it's simpler if you just write to the stdout, with if I recall correctly is this (sorry I should have named this something more obvious like a `msg.Printf()` or something).

```go
			err = kernel.PublishWriteStream(msg, kernel.StreamStdout,
				fmt.Sprintf("%s\n", strings.Join(lines[1:last_gx_line+1], "\n"))
```

Since the end of the range in the slice is "open" (meaning it's one past the last item), you want `last_gx_line+1`, I think.